cmake/modules/LLDBStandalone.cmake

```diff
@@ -50,33 +50,26 @@ if (CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
   find_program(lit_full_path ${lit_file_name} ${config_dirs} NO_DEFAULT_PATH)
   set(LLVM_DEFAULT_EXTERNAL_LIT ${lit_full_path} CACHE PATH "Path to llvm-lit")
 
-  if(LLVM_TABLEGEN)
-    set(LLVM_TABLEGEN_EXE ${LLVM_TABLEGEN})
-  else()
-    if(CMAKE_CROSSCOMPILING)
-      set(LLVM_NATIVE_BUILD "${LLDB_PATH_TO_LLVM_BUILD}/NATIVE")
-      if (NOT EXISTS "${LLVM_NATIVE_BUILD}")
-        message(FATAL_ERROR
-          "Attempting to cross-compile LLDB standalone but no native LLVM build
-          found. Please cross-compile LLVM as well.")
-      endif()
-
-      if (CMAKE_HOST_SYSTEM_NAME MATCHES "Windows")
-        set(HOST_EXECUTABLE_SUFFIX ".exe")
-      endif()
-
-      if (NOT CMAKE_CONFIGURATION_TYPES)
-        set(LLVM_TABLEGEN_EXE
-          "${LLVM_NATIVE_BUILD}/bin/llvm-tblgen${HOST_EXECUTABLE_SUFFIX}")
-      else()
-        # NOTE: LLVM NATIVE build is always built Release, as is specified in
-        # CrossCompile.cmake
-        set(LLVM_TABLEGEN_EXE
-          "${LLVM_NATIVE_BUILD}/Release/bin/llvm-tblgen${HOST_EXECUTABLE_SUFFIX}")
-      endif()
+  if(CMAKE_CROSSCOMPILING)
+    set(LLVM_NATIVE_BUILD "${LLDB_PATH_TO_LLVM_BUILD}/NATIVE")
+    if (NOT EXISTS "${LLVM_NATIVE_BUILD}")
+      message(FATAL_ERROR
+        "Attempting to cross-compile LLDB standalone but no native LLVM build
+        found. Please cross-compile LLVM as well.")
+    endif()
+
+    if (CMAKE_HOST_SYSTEM_NAME MATCHES "Windows")
+      set(HOST_EXECUTABLE_SUFFIX ".exe")
```

Comment:
I know you're just merging this, but why do we need this? Can't we use CMAKE_EXECUTABLE_SUFFIX?

Reply:
This code is upstream, so if anything, that needs to change first?

Reply:
I think this was my code originally. Is CMAKE_EXECUTABLE_SUFFIX determiend by CMAKE_HOST_SYSTEM_NAME or CMAKE_SYSTEM_NAME? That's where I was unsure.

But I agree with Davide, needs to change upstream before it makes its way here.

Reply:
Yup, just wanted the raise the issue as I was looking at the code. Looks like it might be needed indeed: https://gitlab.kitware.com/cmake/cmake/issues/17553